src/gt4py/next/ffront/ast_passes/single_static_assign.py

```diff
@@ -104,18 +268,19 @@ def visit_AnnAssign(self, node: ast.AnnAssign) -> ast.AnnAssign:
             node.value = self._rename(node.value)
             node.target = self.visit(node.target)
         elif isinstance(node.target, ast.Name):
-            target_id = node.target.id
+            # An empty annotation always applies to the next assignment.
+            # So we need to use the correct versioning, but also ensure
+            # we restore the old versioning afterwards, because no assignment
+            # actually happens.
```

Comment:
I don't fully understand the details, but this is referring to this case (test_empty_annotated_assign):
```
a = 0
a: int
b = a
```
If you want I take a closer look.

Reply:
Yes, I understood the same core idea but don't really get the details. I think we should understand what the comment really means and rephrase it in a more comprehensible way before merging the PR, otherwise is useless.